### DISCUSSION TITLE
v0.5

### DISCUSSION THREAD

**Discussion by kaixxx** [kaixxx/noScribe#65] [Announcements]
New:
- WebVTT output, allows importing noScribe transcripts into [EXMARaLDA](https://exmaralda.org)
- plain text output (*.txt)
- improved speaker detection: number of speakers can be defined beforehand
- reduced hallucination and looping by adding a VAD filter
- automatic update notification on new releases
- editor opens automatically (can be disabled in advanced options)

Fixes:
- CUDA support now non-beta
- small fixes with hebrew language setting, chinese UI locale, Linux support, etc.

<hr /><em>This discussion was created from the release <a href='https://github.com/kaixxx/noScribe/releases/tag/v0.5'>v0.5</a>.</em>

**Comment by berndmoos** [comment]
Just downloaded and installed the new version on Windows, and tried with a 3 minute audio file. all worked well including the import of the resulting WebVTT file into EXMARaLDA 👍

**Comment by maun200** [comment]
https://drive.switch.ch/index.php/s/EIVup04qkSHb54j?path=%2F

just for the next person looking for the link :)

**Reply by kaixxx** [reply]
Thanks, but better to also read the download instructions here: https://github.com/kaixxx/noScribe#download-and-installation

**Comment by maun200** [comment]
![Screenshot 2024-06-21 214627](https://github.com/kaixxx/noScribe/assets/18380842/7b87eb3e-af3e-4afb-9077-3460a232713b)

I have got this error. Can anyone help me out?

**Reply by kaixxx** [reply]
Do you have a NVIDIA graphics card in your system?
It seems that noScribe is trying to use it (via the library CUDA), but you have downloaded the non CUDA version. 
You can either download and try the CUDA version (which might give you some speed improvements too), or turn of CUDA in the advanced options:
- Locate the file "config.yml" like described here: https://github.com/kaixxx/noScribe#advanced-options
- Close noScribe and change the following lines in config.yml: Set "pyannote_xpu" and "whisper_xpu" to "cpu".
- Save config.yml and restart noScribe. It should now run without trying to use CUDA. 

(There is still a bug somewhere, the non CUDA version should run without these workarounds. I will look into it.)

**Reply by maun200** [reply]
Oky thanks :) I'll try this tomorrow. Unfortunately, when I try to install the CUDA version, I get an NSIS error on both version 0.4.5 and 0.5. Does anyone have any ideas to fix this?

**Reply by kaixxx** [reply]
What kind of error do you get? It works for me (Windows 11). 

Note that the installer consists of two files: "noScribe_setup_0_5_cuda.exe" (330 KB) and "noScribe_setup_0_5_cuda.nsisbin" (6.1 GB). You need both in the same folder.

**Comment by tara396** [comment]
![IMAGE 2024-08-03 09:54:52](https://github.com/user-attachments/assets/d48dad25-88d9-4ef3-8587-b5062088fd07)

Do you know what I can do? I already tried all of your versions. I have a Macair13 Intel Core I5. I would appreciate any recommendation

**Reply by gernophil** [reply]
You need to follow the instructions given in the readme to allow noScribe to run.

Copied from the homepage:

- Note: Unfortunately, we are currently not able to sign the x86_64 package correctly, so you will get a warning that noScribe and noScribeEdit are from unregistered developers. You have to manually allow noScribe and noScribeEdit to be executed, if your Gatekeeper is active. Follow these steps:
  - Double-click the downloaded dmg-file.
Drag noScribe and noScribeEdit into the link to your applications folder (labeled "drag both here to install")
  - Start noScribe by double-clicking the app within your applications folder. You will get an error that noScribe is from an unregistered developer. Do the same with the noScribe Editor.
  - Go to Settings -> Privacy and Security -> Scroll down until you see a message stating noScribe was prevented from starting and click "open anyway". Again, do the same with the noScribe Editor.
  - From now on, both programs should start without issues.

**Reply by tara396** [reply]
Thanks:)

**Comment by meeker42** [comment]
Hello, I installed v0.5 yesterday, but unfortunately noScibe crashes a few moments after starting the transcription. Just after whisper is loaded, the message "transcribing" appears, and then I get a crash to desktop. I disabled the speaker detection - if it´s on, it crahes after initializing Pyanote, so it seems. I have Windows 10 running, an older Nvidia graphics (GTX 1080) but no CUDA so far (non-CUDA version). Will try this tonight ... btw: does the CUDA kit overwrite the Nvidia graphics driver? Just asking because I love gaming and I don´t want to spoil my system ... :-) thx, m.

**Reply by kaixxx** [reply]
Hello, 
sounds like a similar issue to this: https://github.com/kaixxx/noScribe/issues/76 Try the suggestions I gave there and please report back if it helped.  

I would try to get the non-CUDA version running first.

**Reply by gernophil** [reply]
Hey @meeker42, this seems similar to this here: https://github.com/kaixxx/noScribe/issues/76

AFAIK CUDA is part of the NVIDIA driver and I am pretty sure, nothing will be overwritten :). I am wondering if these crashes happen because of a combination of a newer CUDA version with older NVIDIA drivers or vice versa, @kaixxx.

**Reply by meeker42** [reply]
#76 is exactly what happens, yes! I already set the variables manually as described, it did not solve it, but ...  
Updating the Nvidia driver could be a changer, thx! Haven´t done this for some time (i did not expect much improvement with the old 1080 board). I will try that!! Did not think about this yesterday ... I will let you know, thank you!!

**Reply by kaixxx** [reply]
If you set both variables to `cpu` in config.ini (attention: noScribe must be closed while doing this!), CUDA should not be involved at all. So I suspect that we have another problem here. But without a proper error message, it is very hard to imagine what could be the issue.

Do you have any kind of anti virus program installed? Maybe this is interfering with noScribe or the noScribe installer.

**Reply by meeker42** [reply]
It´s Norton 360. Tried this out and deacitvated it, but no effect. Norton stayed quiet as well ... I checked the .exe files and "told" Norton to let go and just trust them and me :-) Installation was smooth so far.

**Reply by kaixxx** [reply]
I haven't heard about problems with Norton and noScribe yet. But it is possible that Norton prevented certain elements of noScribe to be installed correctly. You could try reinstalling noScribe with Norton deactivated.

**Reply by meeker42** [reply]
okay, I will try this as well. Thanks! I see that it´s possible to install to a different drive/folder than C:/... 
do you recommend to keep it on C: or is it no factor?

**Reply by kaixxx** [reply]
It should be no problem installing noScribe on another drive. But I would use the default settings for now (if possible) until we've found the cause of the current issue.

**Reply by meeker42** [reply]
Hello and good morning, sorry to say that none of it worked. Somehow something seems to be blocking the transcription process, ctd after initializing whisper. Nvidia drivers are up to date, Norton was switched off during installation in C:/ and during transcription ... but we tried, thx for you support!

**Reply by kaixxx** [reply]
It annoys me a little that we didn't find the issue, but I'm out of ideas atm. Here you'll find older versions of noScribe if you want to try: https://drive.switch.ch/index.php/s/EIVup04qkSHb54j

**Reply by meeker42** [reply]
yes ... I am a bit down, too. Thought that the Nvidia update or/and Norton off would do the trick ... 
thank you for your quick response :-)
if something changes, I will let you know! m.

**Reply by gernophil** [reply]
How about the rest of you system? Is this up to date? Windows updates, chipset drivers, audio drivers etc. I know this issue from other programs and especially games. It's always hard to find the reason for this. You could also try to start windows in safe mode and see, if it works there.

**Reply by meeker42** [reply]
Safe mode, yes. Worth a try :-) everything else updated so far.

**Comment by Lod3** [comment]
Is there a way to host the file on a faster file hoster? It takes an hour to download 6GB and It failed 2 times. It was able to continue without restarting, that was nice though.

**Reply by kaixxx** [reply]
I think one hour for 6GB is not too bad. Two failures however is not good. I will keep an eye on that. Do you know a free hoster which is faster?

**Reply by Lod3** [reply]
I was thinking Proton Drive, but the free tier capacity is only 5GB, can you compress the nsisbin file? Even then compressing it down 1GB is going to be optimistic probably.

**Reply by kaixxx** [reply]
The files are already compressed... 
SwitchDrive proved to be quite reliable in the past. Today, quite a few people were downloading noScribe at the same time. Maybe that's part of the reason why it was a little slow today.

**Comment by Ilya-77** [comment]
Hello!
Version 0.5.
Would you please to explain cuda using 2 GPUs?
I have 2 GPUs Tesla P40 but noScribe  always uses only one of them.
Is it possible to use noScribe for both GPUs at the same time?

**Reply by kaixxx** [reply]
I have no idea, tbh. I don't even own a single CUDA enabled GPU myself... 
Here is a long thread where people are discussing their experiences with CUDA: https://github.com/kaixxx/noScribe/discussions/50 Some had good luck with running several transcriptions in parallel (that is, several instances of noScribe at the same time). You can try that. 

Otherwise, you'll have to run noScribe directly from source and experiment yourself with the settings for CUDA.

**Reply by gernophil** [reply]
According to [this](https://stackoverflow.com/a/64825728/16611809) `torch.device('cuda')` (IIRC that's what noScribe does) should use all GPUs. Maybe you have to set `CUDA_VISIBLE_DEVICES` (in your case, I'd assume to `0,1`: `CUDA_VISIBLE_DEVICES=0,1`). For this to have an effect on noScribe, you would need to run it from the command-line with this env variable set, but I have no experience on doing this in windows.

**Reply by Ilya-77** [reply]
Ok, thanks for reply and information 🙏

**Reply by Ilya-77** [reply]
> According to [this](https://stackoverflow.com/a/64825728/16611809) `torch.device('cuda')` (IIRC that's what noScribe does) should use all GPUs. Maybe you have to set `CUDA_VISIBLE_DEVICES` (in your case, I'd assume to `0,1`: `CUDA_VISIBLE_DEVICES=0,1`). For this to have an effect on noScribe, you would need to run it from the command-line with this env variable set, but I have no experience on doing this in windows.

Thanks.

**Reply by gernophil** [reply]
Did you have any look trying this, @Ilya-77?

**Reply by Ilya-77** [reply]
> Did you have any look trying this, @Ilya-77?

Yes I did, but I have windows 2022 server and I've no idea how to set up both GPUs using [this](https://stackoverflow.com/a/64825728/16611809) advice.
